### PR TITLE
Ignore build_docs.lock

### PR DESCRIPTION
**Pull request by egeakman**

### CODE REVIEW

**Review by AA-Turner — APPROVED** [review]

**Review by willingc — APPROVED** [review]

**Review by ezio-melotti — APPROVED** [review]
The change seems harmless enough, but where does this file come from?

**Comment by egeakman** [comment]
> The change seems harmless enough, but where does this file come from?

It is created at [line 1158](https://github.com/python/docsbuild-scripts/blob/56d72d43e5759cc0ed600827b56e81d8310bcaca/build_docs.py#L1158).

This was added years ago, but I don't know when the file started to show up and not cleaned.

**Comment by hugovk** [comment]
On the server, the script is started by an hourly cron schedule. The script checks for the lockfile, indicating there's already a running build. 

If so, the new one exits, allowing the current one to continue. 

If not, it creates a lockfile and begins the 40 hour build.

File: .gitignore [hunk 1]
```diff
@@ -1,6 +1,7 @@
 /build_root/
 /logs/
 /www/
+build_docs.lock
```
Comment (ezio-melotti) [inline .gitignore:4]:
```suggestion
# temporary lock file created while building the docs
build_docs.lock
```

Maybe add a comment like this?

**Review by willingc — APPROVED** [review]

**Pull request merged**